incubator/hnc/hack/ci-test.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -e
+
+export PATH=$(go env GOPATH)/bin:$PATH
```

Comment:
Why is this part necessary?

Reply:
I based this script on this script: https://github.com/kubernetes-sigs/controller-runtime/blob/master/hack/ci-check-everything.sh#L22, however, I haven't tried without the `export ...` to see if it fails. I'll try it (shouldn't have added this without knowing for sure why it is used)

Reply:
You should be able to test this whole thing in the `golang` container. Do you know how to run this script in the golang Docker container?

Reply:
Yup, I need to mount the dir in order to access this file inside the container. 

In Prow, this script will be run inside a docker container with the golang image. I'll send a commit removing this export and test it locally with Prow. I need to send a commit because Prow uses a PR number and its current state to run locally.

Reply:
I'm not sure I fully understand that - you mean that when you pushed this commit, you're somehow getting Prow to run? Where? On your forked repo? Are you running it locally? Where are the instructions you're following? Inquiring minds want to know! :)

Reply:
Yup until now I thought that [Prow locally](https://github.com/kubernetes/test-infra/blob/master/prow/build_test_update.md#example) used the current state of the opened PR (it asks for the PR to clone when running locally), which would be this one to run the job, but I just noticed that this is not the case, hahahahhaa :sweat_smile:. It clones the repo and uses the master branch to test it locally. I did an `ls -R` on the job commands and this file, `ci-test.sh`, was not present on the output.

So summing up, I think this PR needs to merged in order for me to test it locally with Prow. Not sure I if explained myself correctly...

Reply:
From the clone log, I don't really understand why this file is not present, because it merges this PR with the master. I'm trying to understand.

Reply:
But anyways, I can test this file alone and then when this PR is merged. I can test locally with Prow. The code for the test-infra repo is done already, I'll open there after this one is merged :)

Reply:
Hmm. Have you been able to at least run this script manually in a docker container? i.e., without Prow? If it doesn't work w/o Prow, it definitely won't work _with_ it.

```
docker run -it golang /bin/bash
<manually get the code>
hack/ci-test.sh
<see if it works>
```

Reply:
Also, maybe comment out these `PATH` modifying lines for now if you're not sure they're needed. Or leave them in if the script doesn't run without them!

Reply:
Yup, I was able to run inside a container. I removed those PATH lines and everything worked.

Reply:
Ok, either delete them or comment them out for now. And update the commit message (if you haven't already) to point out how you tested this script.

Reply:
Okidoki :smile: